In [1]:
import pandas as pd
import numpy as np
import scipy.stats as stats

In [2]:
df = pd.read_csv('weight-height - weight-height.csv')
df.head(2)

Gender     Height      Weight
0   Male  73.847017  241.893563
1   Male  68.781904  162.310473

In [3]:
females = df.loc[ df['Gender'] == 'Female'].copy()

In [4]:
data = females['Height']

In [5]:
#set all the important stats, variables,and the range
low_end = 55.0
high_end = 56.0
mean = np.mean(data)
std = np.std(data)
min = np.min(data)
max = np.max(data)
x_range = np.linspace(min, max,1000)

In [6]:
#plot the normal CDF
cdf = stats.norm.cdf(x_range, loc=mean, scale=std)
print(stats.norm.cdf(high_end, loc=mean,scale=std) - stats.norm.cdf(low_end,loc=mean, scale=std))

0.0015043054289559745


The probability of a female's height between 55.0 and 56.0 inches is: 1.5%.

In [7]:
print(stats.norm.cdf(60, loc=mean,scale=std))

0.0844649272520746


The probability of a female's height less than 5 feet is: 8.4%.

This is a normal distribution data, So the probability of above or below the mean are both 50%.